<a href="https://colab.research.google.com/github/divyakedia/playstore-data-analysis/blob/main/(Divya_Self)Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

# 1. Importing the libraries necessary for project

In [2]:
#The first step is to import the libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

# 2. Mounting google drive


In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# 3. Creating path variable for the datasets

In [4]:
#creating data files path and reading the csv files
playstore_data_path = "/content/drive/MyDrive/(EDA)Capstone Project/Play Store Data (1).csv"
playstore_df = pd.read_csv(playstore_data_path)
users_review_path = "/content/drive/MyDrive/(EDA)Capstone Project/User Reviews.csv"
users_review_df = pd.read_csv(users_review_path)

# 4. Viewing the data of Python database

# 4.1 PlayStore Data

In [14]:
#Viewing the dataframe of playstore data
playstore_df.head(-1)

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10835                                           FR Forms             BUSINESS   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10835     NaN       0                9.6M          10+  Free     0   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   

      Content Rating                     Genres        Last Updated  \
0           Everyone               Art & Design     January 7, 2018   
1           Everyone  Art & Design;Pretend Play    January 15, 2018   
2           Everyone               Art & Design      August 1, 2018   
3               Teen               Art & Design        June 8, 2018   
4           Everyone    Art & Design;Creativity       June 20, 2018   
...              ...                        ...                 ...   
10835       Everyone                   Business  September 29, 2016   
10836       Everyone                  Education       July 25, 2017   
10837       Everyone                  Education        July 6, 2018   
10838       Everyone                    Medical    January 20, 2017   
10839     Mature 17+          Books & Reference    January 19, 2015   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10835               1.1.5          4.0 and up  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  

[10840 rows x 13 columns]

In [17]:
playstore_df.shape

(10841, 13)

In [6]:
# Print last 5 no. of rows
playstore_df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [7]:
# Printing basic information about the DataFrame
playstore_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [8]:
# To view available numeric column
playstore_df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

In [10]:
# To view all the columns of the available dataframe
playstore_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

# 4.2 User Review Data

In [11]:
users_review_df

App  \
0            10 Best Foods for You   
1            10 Best Foods for You   
2            10 Best Foods for You   
3            10 Best Foods for You   
4            10 Best Foods for You   
...                            ...   
64290  Houzz Interior Design Ideas   
64291  Houzz Interior Design Ideas   
64292  Houzz Interior Design Ideas   
64293  Houzz Interior Design Ideas   
64294  Houzz Interior Design Ideas   

                                       Translated_Review Sentiment  \
0      I like eat delicious food. That's I'm cooking ...  Positive   
1        This help eating healthy exercise regular basis  Positive   
2                                                    NaN       NaN   
3             Works great especially going grocery store  Positive   
4                                           Best idea us  Positive   
...                                                  ...       ...   
64290                                                NaN       NaN   
64291                                                NaN       NaN   
64292                                                NaN       NaN   
64293                                                NaN       NaN   
64294                                                NaN       NaN   

       Sentiment_Polarity  Sentiment_Subjectivity  
0                    1.00                0.533333  
1                    0.25                0.288462  
2                     NaN                     NaN  
3                    0.40                0.875000  
4                    1.00                0.300000  
...                   ...                     ...  
64290                 NaN                     NaN  
64291                 NaN                     NaN  
64292                 NaN                     NaN  
64293                 NaN                     NaN  
64294                 NaN                     NaN  

[64295 rows x 5 columns]

In [12]:
# To view available numeric column
users_review_df.describe()

Sentiment_Polarity  Sentiment_Subjectivity
count        37432.000000            37432.000000
mean             0.182146                0.492704
std              0.351301                0.259949
min             -1.000000                0.000000
25%              0.000000                0.357143
50%              0.150000                0.514286
75%              0.400000                0.650000
max              1.000000                1.000000

In [13]:
# Printing basic information about the DataFrame
users_review_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


# 5. Cleaning the data of Playstore Dataframe

By Studying the DataFrame, we see noticed that,


*  The Shape of the DataFrame is (10841, 13).
*    Column 'Reviews', 'Size', 'Installs' and 'Price' are in the type of 'object'.
*   Column 'Size' have the  strings representing size in 'M' as Megabytes, 'k' as kilobytes and also 'Varies with devices'.
*   Column 'Installs' have strings representing no. of installs with symbols such as ',' and '+'.
*   Column 'Price' have strings representing price with symbol '$'.

Thus, the data has to be cleaned for further process.





In [62]:
# Find the row with insufficiant data:
playstore_df[playstore_df['Type']!='Free'][playstore_df[playstore_df['Type']!='Free']['Price']=='0']


Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [63]:
#Drop the row from the data frame:
playstore_df.drop(playstore_df[playstore_df['Type']!='Free'][playstore_df[playstore_df['Type']!='Free']['Price']=='0'].index, inplace=True)

In [64]:
#Find value mismatched row:
playstore_df[playstore_df['Genres']=='February 11, 2018']

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [65]:
#Find the index of the row which contains the carbage value:
playstore_df[playstore_df['Genres']=='February 11, 2018'].index

Int64Index([], dtype='int64')

In [66]:
#Drop the found carbage row from our dataframe:
playstore_df.drop(playstore_df[playstore_df['Genres']=='February 11, 2018'].index, inplace=True)

In [70]:
#making a fuction to clean the strings:
def clean_data(val):

  
  if 'M' in str(val):
    val = str(int(float(val.replace('M',''))*1000000))
  if 'k' in str(val):
    val = str(int(float(val.replace('k',''))*1000))
  if '+' in str(val):
    val = val.replace('+','')
  if ',' in str(val):
    val = val.replace(',','')
  if 'NaN' in str(val):
    val = '0'
  if '$' in str(val):
    val = val.replace('$','')
  else:
    pass
  return val

In [71]:
# Cleaning the unwanted charactors and converting the required column values into valid numeric type

#Changing the 'Reviews' column values into valid numeric values
playstore_df['Reviews'] = pd.to_numeric(playstore_df['Reviews'])

#Changing the 'Size' column values into valid numeric values
playstore_df['Size'] = playstore_df['Size'].apply(lambda x: str(x).replace('Varies with device', 'NaN') if 'Varies with device' in str(x) else x)
playstore_df['Size'] = pd.to_numeric(playstore_df['Size'].map(lambda x: clean_data(x)))

#Changing the 'Installs' column values into valid numeric values
playstore_df['Installs'] = pd.to_numeric(playstore_df['Installs'].map(lambda x: clean_data(x)))

#Changing the 'Price' column values into valid numeric values
playstore_df['Price'] = pd.to_numeric (playstore_df['Price'].map(lambda x:clean_data(x)))
playstore_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10839 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10839 non-null  object 
 1   Category        10839 non-null  object 
 2   Rating          9366 non-null   float64
 3   Reviews         10839 non-null  float64
 4   Size            10839 non-null  int64  
 5   Installs        10839 non-null  int64  
 6   Type            10839 non-null  object 
 7   Price           10839 non-null  float64
 8   Content Rating  10839 non-null  object 
 9   Genres          10839 non-null  object 
 10  Last Updated    10839 non-null  object 
 11  Current Ver     10831 non-null  object 
 12  Android Ver     10837 non-null  object 
dtypes: float64(3), int64(2), object(8)
memory usage: 1.4+ MB
